astropy/io/fits/hdu/base.py

```diff
@@ -855,16 +855,7 @@ def match_header(cls, header):
         """
         # The SIMPLE keyword must be in the first card
         card = header.cards[0]
-
-        # The check that 'GROUPS' is missing is a bit redundant, since the
-        # match_header for GroupsHDU will always be called before this one.
-        if card.keyword == "SIMPLE":
-            if "GROUPS" not in header and card.value is False:
-                return True
-            else:
-                raise InvalidHDUException
-        else:
-            return False
+        return card.keyword == "SIMPLE" and card.value is False
```

Comment:
This change is not strictly needed since `_NonstandardHDU` comes after the other main extensions when finding the correct HDU class, but as the comment says this check is useless.